akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -911,11 +911,67 @@ trait FlowOps[+Out, +Mat] {
+   * an array up to the allowed max limit if the upstream publisher is faster.
+   *
+   * This element only rolls up elements if the upstream is faster, but if the downstream is faster it will not
+   * duplicate elements.
+   *
+   * '''Emits when''' downstream stops backpressuring and there is an aggregated element available
+   *
+   * '''Backpressures when''' there are `max` batched elements and 1 pending element and downstream backpressures
+   *
+   * '''Completes when''' upstream completes and there is no batched/pending element waiting
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * See also [[FlowOps.conflate]], [[FlowOps.batchWeighted]]
+   *
+   * @param max maximum number of elements to batch before backpressuring upstream (must be positive non-zero)
+   * @param seed Provides the first state for a batched value using the first unconsumed element as a start
+   * @param aggregate Takes the currently batched value and the current pending element to produce a new aggregate
+   */
+  def batch[S](max: Long, seed: Out ⇒ S)(aggregate: (S, Out) ⇒ S): Repr[S] =
+    via(Batch(max, seed, aggregate))
+
+  /**
+   * Allows a faster upstream to progress independently of a slower subscriber by aggregating elements into batches
+   * until the subscriber is ready to accept them. For example a batch step might concatenate `ByteString`
+   * elements up to the allowed max limit if the upstream publisher is faster.
+   *
+   * This element only rolls up elements if the upstream is faster, but if the downstream is faster it will not
+   * duplicate elements.
+   *
+   * Batching will apply for all elements, even if a single element cost is greater than the total allowed limit.
+   * In this case, previous batched elements will be emitted, then the "heavy" element will be emitted (after
+   * being applied with the `seed` function) without batching further elements with it, and then the rest of the
+   * incoming elements are batched.
+   *
+   * '''Emits when''' downstream stops backpressuring and there is a batched element available
+   *
+   * '''Backpressures when''' there are `max` weighted batched elements + 1 pending element and downstream backpressures
+   *
+   * '''Completes when''' upstream completes and there is no batched/pending element waiting
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * See also [[FlowOps.conflate]], [[FlowOps.batch]]
+   *
+   * @param max maximum weight of elements to batch before backpressuring upstream (must be positive non-zero)
+   * @param costFn a function to compute a single element weight
+   * @param seed Provides the first state for a batched value using the first unconsumed element as a start
+   * @param aggregate Takes the currently batched value and the current pending element to produce a new batch
+   */
+  def batchWeighted[S](max: Long, costFn: Out ⇒ Long, seed: Out ⇒ S)(aggregate: (S, Out) ⇒ S): Repr[S] =
```

Comment:
I think this can fully replace conflate’s use-cases. Maybe we should make `conflate == batchWeighted(1, _ => 0, identity)`?